planex/makesrpm.py

```diff
@@ -0,0 +1,132 @@
+import sys
+import subprocess
+import os
+from shutil import copyfile
+import fileinput
+import tarfile
+import tempfile
+import shutil
+
+import argparse
+import argcomplete
+from planex.util import add_common_parser_options
+
+
+def parse_args_or_exit(argv):
+    """
+    Parse command line options
+    """
+    parser = argparse.ArgumentParser(description='Planex SRPM building')
+    add_common_parser_options(parser)
+    parser.add_argument(
+        "--topdir", metavar="DIR", default=None,
+        help='Set rpmbuild toplevel directory')
+    parser.add_argument(
+        "--dist", metavar="DIST", default="",
+        help="distribution tag (used in RPM filenames)")
+    argcomplete.autocomplete(parser)
+    return parser.parse_known_args(argv)
+
+
+def parse_patchseries(series_file):
+    """
+    Parse series file and return the list of patches
+    """
+    patches = []
+    for line in open(series_file):
+        li = line.strip()
+        if not li.startswith("#"):
+            patches.append(li)
+
+    return patches
+
+
+def main(argv):
+    """
+    Entry point arguments: component.spec Source0 Source1 ... Patch0 Patch1 ...
+    The specfile must always be present and it has to be the first in the list
+    """
+    intercepted_args, passthrough_args = parse_args_or_exit(argv)
+    target = os.path.basename(passthrough_args[0]).split('.')[0]
+    specfile = passthrough_args[0]
+    tmp_dirpath = tempfile.mkdtemp() + '/'
+    tmp_specs = tmp_dirpath + 'SPECS/'
+    tmp_specfile = tmp_dirpath + specfile
+    tmp_build = tmp_dirpath + '_build/'
+    tmp_sources = tmp_build + 'SOURCES/'
+
+    # Create temporary working area
+    os.makedirs(tmp_specs)
+    os.makedirs(tmp_build)
+    os.makedirs(tmp_sources)
+
+    try:
+        # Copy files to temporary working area
```

Comment:
I think it would be good to the logic that sets up the temporary working area into a separate function.